openai-chat.el

```diff
@@ -60,7 +60,7 @@ STREAM, STOP, MAX-TOKENS, PRESENCE-PENALTY, FREQUENCY-PENALTY, and LOGIT-BIAS."
   (openai-request "https://api.openai.com/v1/chat/completions"
     :type "POST"
     :headers `(("Content-Type"  . "application/json")
-               ("Authorization" . ,(concat "Bearer " key)))
+               ("Authorization" . ,(concat "Bearer " (openai--resolve-key key))))
```

Comment:
Ok, I also note that this would be required in the other repos that depend on openai, are you amenable to similar PRs in those repos

Reply:
I think perhaps moving the creation of the default headers into `openai-request` might be bettter, so the `:headers` could be removed from all those callsites ? I guess any headers that _were_ specified could be merged on top.

Reply:
> Ok, I also note that this would be required in the other repos that depend on openai, are you amenable to similar PRs in those repos

Other packages fully rely on this package, I don't think we will need to change anything from those packages.

> I think perhaps moving the creation of the default headers into `openai-request` might be bettter, so the `:headers` could be removed from all those callsites ? I guess any headers that were specified could be merged on top.

This sounds reasonable. Some endpoints (mostly `GET`) don't require the key; but now, every endpoint seems to require it. 😕 As of right now, I might prefer to leave it as it is since I do not know when they are going to change it (require key or don't require key). WDYT?

Reply:
> Ok, I also note that this would be required in the other repos that depend on openai, are you amenable to similar PRs in those repos

The call to resolve key should move to inside openai-request - then no call sites need to change.